packages/twenty-front/src/modules/settings/accounts/components/SettingAccountBlocklistContactRow.tsx

```diff
@@ -0,0 +1,287 @@
+import styled from '@emotion/styled';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useContext, useEffect, useState } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { Key } from 'ts-key-enum';
+import { z } from 'zod';
+
+import { BlocklistItem } from '@/accounts/types/BlocklistItem';
+import { StyledInput } from '@/object-record/object-filter-dropdown/components/ObjectFilterDropdownFilterSelect';
+import { BLOCKLIST_CONTEXT_DROPDOWN_ID } from '@/settings/accounts/constants/BlocklistContextDropdownId';
+import { BlocklistContext } from '@/settings/accounts/contexts/BlocklistContext';
+import { BlocklistContactLevel } from '@/settings/accounts/types/BlocklistContactLevel';
+import { TextInput } from '@/ui/input/components/TextInput';
+import { Dropdown } from '@/ui/layout/dropdown/components/Dropdown';
+import { DropdownMenuItemsContainer } from '@/ui/layout/dropdown/components/DropdownMenuItemsContainer';
+import { IconChevronDown, IconTrash, MenuItemMultiSelect } from 'twenty-ui';
+import { isDomain } from '~/utils/is-domain';
+import { isDefined } from '~/utils/isDefined';
+
+type SettingsAccountsBlocklistContactRowProps = {
+  item?: BlocklistItem;
+};
+
+const StyledContainer = styled.div`
+  align-items: center;
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  gap: ${({ theme }) => theme.spacing(2)};
+`;
+
+const StyledLinkContainer = styled.div`
+  flex: 1;
+`;
+
+const StyledEmptyBox = styled.div`
+  width: ${({ theme }) => theme.spacing(8)};
+`;
+
+const StyledRemoveButton = styled(IconTrash)`
+  color: ${({ theme }) => theme.font.color.light};
+  cursor: pointer;
+  height: ${({ theme }) => theme.spacing(4)};
+  width: ${({ theme }) => theme.spacing(4)};
+  margin-left: ${({ theme }) => theme.spacing(2)};
+  margin-right: ${({ theme }) => theme.spacing(2)};
+`;
+
+const StyledClickableComponent = styled.div`
+  align-items: center;
+  cursor: pointer;
+  display: flex;
+  justify-content: space-between;
+  position: relative;
+`;
+
+const StyledInputButton = styled(TextInput)`
+  border: 1px solid ${({ theme }) => theme.border.color.light};
+  border-radius: 4px;
+  z-index: -1;
+`;
+
+const StyledIconChevronDown = styled(IconChevronDown)`
+  color: ${({ theme }) => theme.font.color.light};
+  position: absolute;
+  height: 16px;
+  width: 16px;
+  right: 8px;
+  top: 50%;
+  transform: translateY(-50%);
+`;
+
+const blocklistLevelDropdownItems: {
+  id: BlocklistContactLevel;
+  label: string;
+}[] = [
+  {
+    id: BlocklistContactLevel.FROMTO,
+    label: 'From/To',
+  },
+  {
+    id: BlocklistContactLevel.CC,
+    label: 'Cc',
+  },
+  {
+    id: BlocklistContactLevel.BCC,
+    label: 'Bcc',
+  },
+];
+
+const computeBlocklistItemHandle = (
```

Comment:
I don't understand this function looking at the naming, what are we computing and why it returns a boolean? Could you clarify?

Reply:
This is no longer needed in the validation, pushing updates asap